AutomatedTesting/Gem/PythonTests/editor/EditorScripts/ComponentCRUD_Add_Delete_Components.py

```diff
@@ -100,7 +101,7 @@ async def add_component(component_name):
         general.select_object("Entity1")
 
         # Give the Entity Inspector time to fully create its contents
-        general.idle_wait(0.5)
+        general.idle_wait_frames(3)
```

Comment:
Blind waits are a pain to debug. Frames should be more stable than seconds, but is there another signal you can check to verify whether the contents have been loaded/created? Is that the purpose of the mesh component?

Reply:
Unfortunately, there isn't. Discussed using entity_inspector.updatesEnabled(), but that apparently doesn't guarantee that things are set to where we need them for UI Interactions.

Reply:
you could get the read-only mesh component properties for Mesh Stats which is a container property. if the mesh component doesn't have a model specified the container would be empty. once the model is specified it will contain at least one element 'LOD 0' which would have non-zero mesh count and vertex count. you could wait_for_condition and check that Mesh Stats has a non-zero container count. I'm not sure if that would satisfy your needs though

Reply:
It's the actual Qt widgets we need to be ready for this particular test. The Add Component functionality and later components are probably "ready" a lot quicker than this, but the test simulates user input as closely as it can, so needs the actual widgets to be drawn and ready for interaction, and there's no signal for that (that I'm aware of, unless entity_inspector.updatesEnabled() is expected to be reliable to indicate that--there's some planned discussion around that with some Qt experts as it's also planned to be used for measuring some performance metrics down the line).